apps/api/src/app/modules/applications/applications.service.ts

```diff
@@ -17,14 +17,49 @@ export class ApplicationsService {
     const result = await this.connection
       .collection(this.taskModel.collection.collectionName)
       .aggregate<Application>([
+        // Get only the last seven days using StartDate
+        {
+          $match: {
+            $expr: {
+              $gte: [
+                '$StartDate',
+                new Date(Date.now() - 7 * 24 * 60 * 60 * 1000),
+              ],
+            },
+          },
+        },
         {
           // Groupe by Options.Options.GridAppName and sum tasks using Status
           $group: {
-            _id: '$Options.Options.GridAppName',
+            _id: {
+              applicationName: '$Options.Options.GridAppName',
+              applicationVersion: '$Options.Options.GridAppVersion',
+            },
+            countTasksPending: {
+              $sum: {
+                $cond: {
+                  if: {
+                    $or: [
```

Comment:
Voir pour faire une variable plus globale avec un $in 

https://www.mongodb.com/docs/manual/reference/operator/query/in/